src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -71,34 +73,51 @@ class KNN80DocValuesConsumer extends DocValuesConsumer implements Closeable {
     @Override
     public void addBinaryField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
         delegatee.addBinaryField(field, valuesProducer);
-        if (field.attributes().containsKey(KNNVectorFieldMapper.KNN_FIELD)) {
+        if (isKNNBinaryFieldRequired(field)) {
             addKNNBinaryField(field, valuesProducer);
         }
     }
 
-    public void addKNNBinaryField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
+    private boolean isKNNBinaryFieldRequired(FieldInfo field) {
+        final KNNEngine knnEngine = getKNNEngine(field);
+        log.info(String.format("Selected engine [%s] for field [%s]", knnEngine.getName(), field.getName()));
```

Comment:
```Info``` might be spamming logs while indexing. Should we move to ```debug```? Same in other places as well.

Reply:
correct. This part of code gets called for every refresh during indexing. By default refresh happens every 1second.  This could be too much logging. Thanks for fixing